exporters/webhook/plugins/pelorus_handler.py

```diff
@@ -35,10 +38,100 @@
     HTTPException,
     PelorusWebhookPlugin,
     PelorusWebhookResponse,
-    Request,
 )
 
 
+def _verify_payload_signature(
+    secret: bytes, signature_secret: str, json_payload_data: Dict[str, str]
+) -> bool:
+    """
+    This function attempts to match the hash of a payload to its data,
+    with the understanding that the input JSON may be formatted slightly
+    differently, such as having different separators or newlines.
+
+    Any change to a single character in the sender's JSON input will
+    change its hash signature. Therefore, on the webhook side,
+    this function attempts to ignore separators and newline characters
+    in order to match the incoming hash using variations of those formats
+    that do not affect the payload data.
+
+    To illustrate this point, consider the following variations of
+    the same JSON input, which are actually equivalent:
+
+    {"data": "value", "data2": "value2"}
+    {"data":"value","data2":"value2"}
+    {"data":"value", "data2":"value2"}
+    {"data" :"value","data2" :"value2"}
+    {"data" :"value", "data2" :"value2"}
+    {"data": "value", "data2": "value2"}\n
+    {"data":"value","data2":"value2"}\n
+    {"data":"value", "data2":"value2"}\n
+    {"data" :"value","data2" :"value2"}\n
+    {"data" :"value", "data2" :"value2"}\n
+    { "data": "value", "data2": "value2" }
+    { "data":"value","data2":"value2" }
+    { "data":"value", "data2":"value2" }
+    { "data" :"value","data2" :"value2" }
+    { "data" :"value", "data2" :"value2" }
+    { "data": "value", "data2": "value2" }\n
+    { "data":"value","data2":"value2" }\n
+    { "data":"value", "data2":"value2" }\n
+    { "data" :"value","data2" :"value2" }\n
+    { "data" :"value", "data2" :"value2" }\n
+
+    Returns:
+        bool: True when the matching hash was found, False otherwise
+    """
+
+    # Formats:
+    #  {"data": "value", "data2": "value2"}
+    #  {"data":"value","data2":"value2"}
+    #  {"data":"value", "data2":"value2"}
+    #  {"data" :"value","data2" :"value2"}
+    #  {"data" :"value", "data2" :"value2"}
+    separator_formats = [
+        (", ", ": "),
+        (",", ":"),
+        (", ", ":"),
+        (",", " :"),
+        (", ", " :"),
+    ]
+    additional_transformation = [
+        "extra_spaces_no_newline",
+        "extra_spaces_newline",
+        "no_extra_spaces_no_newline",
+        "no_extra_spaces_newline",
+    ]
+
+    for separator in separator_formats:
+        for idx, transformation in enumerate(additional_transformation):
+            # no_extra_spaces_no_newline - default
+            payload_json_string = json.dumps(
+                json_payload_data, separators=separator, indent=None
+            )
+            payload_json_string.rstrip("\n")
+
+            if transformation == "extra_spaces_no_newline":
```

Comment:
I think that would make things less visible/clear by reading the code.

Reply:
line 106 is unnecessary? I commented it and no test broke

How it would like with the change I commented before
```python
    separator_formats = [
        (", ", ": "),
        (",", ":"),
        (", ", ":"),
        (",", " :"),
        (", ", " :"),
    ]
    transformations = [
        lambda text: text,  # default
        lambda text: text + "\n",
        lambda text: f"{{ {text[1:-1]} }}",
        lambda text: f"{{ {text[1:-1]} }}\n",
    ]

    for separator in separator_formats:
        for transformation in transformations:
            payload_json_string = transformation(
                json.dumps(json_payload_data, separators=separator, indent=None)
            )

            encoded_payload = payload_json_string.encode("utf-8")

            sha256_signature = (
                "sha256="
                + hmac.new(secret, encoded_payload, hashlib.sha256).hexdigest()
            )

            if hmac.compare_digest(sha256_signature, signature_secret):
                return True
    return False
```

Reply:
Line 106 is necessary for  the first case 'extra_spaces_no_newline' so we remove newline if was there.

I prefer to not not have such nested function, as it may be harder to debug imo.

Reply:
then why no tests broke when we comment it?

Reply:
I think `json.dumps` (with the parameters we use) do not create string with line break, and that's why the tests are not breaking (and line 106 is not necessary)

Reply:
Maybe, but it doesn't hurt to add one to not be dependent on the json.dumps() return format. There is nothing in the documentation of dump() nor dumps() which says this is the case.